flink-state-backends/flink-statebackend-changelog/src/main/java/org/apache/flink/state/changelog/ChangelogKeyedStateBackend.java

```diff
@@ -694,6 +705,10 @@ public KeyedStateBackend<K> getDelegatedKeyedStateBackend(boolean recursive) {
         return keyedStateBackend.getDelegatedKeyedStateBackend(recursive);
     }
 
+    public ChangelogMetricGroup getMetrics() {
+        return metrics;
+    }
```

Comment:
> the backend would only need report these sizes; and Materializer - events, such as started/completed/failed materialization.

Maybe I could move completed into materialization part, but the precise place about starting materilization is `initMaterialization` where I report currently (From starting sync materlization on).
 
Or maybe we could just report started before `initMaterialization` even if it doesn't have done a real materilization ?

Reply:
If I understand correctly, you are talking about where to call `reportPendingMaterialization()`: `ChangelogKeyedStateBackend.initMaterialization` or `PeriodicMaterializationManager.triggerMaterialization`.
I think it doesn't matter much in terms of usefulness (on one hand, it might be useful to know whether the materialization was scheduled or not at all; OTH, `mailboxExecutor` might delay it after triggering).
So I think the latter (from `PeriodicMaterializationManager`) is fine.